src/UI/Areas/Identity/Services/ExternalLoginsServices.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Identity.UI.Services
+{
+    public abstract class ExternalLoginsServices
```

Comment:
This seems kinda yucky to have to do this, basically wrapping a call to signInManager in an abstract class

Reply:
Yeah, It’s a limitation due to the lack of open genetics. I’m gonna try and figure out a better solution. Moving forward avoid injecting anything depending on TUser into the view